homeassistant/components/climacell/config_flow.py

```diff
@@ -0,0 +1,163 @@
+"""Config flow for ClimaCell integration."""
+import logging
+from typing import Any, Dict
+
+from pyclimacell import ClimaCell
+from pyclimacell.const import REALTIME
+from pyclimacell.exceptions import (
+    CantConnectException,
+    InvalidAPIKeyException,
+    RateLimitedException,
+)
+import voluptuous as vol
+
+from homeassistant import config_entries, core
+from homeassistant.const import CONF_API_KEY, CONF_LATITUDE, CONF_LONGITUDE, CONF_NAME
+from homeassistant.core import callback
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import (
+    CONF_FORECAST_TYPES,
+    CONF_TIMESTEP,
+    DAILY,
+    DEFAULT_FORECAST_TYPE,
+    DEFAULT_NAME,
+    DEFAULT_TIMESTEP,
+    HOURLY,
+    NOWCAST,
+)
+from .const import DOMAIN  # pylint: disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _get_config_schema(
+    hass: core.HomeAssistant, input_dict: Dict[str, Any] = None
+) -> vol.Schema:
+    """
+    Return schema defaults for init step based on user input/config dict.
+
+    Retain info already provided for future form views by setting them as
+    defaults in schema.
+    """
+    if input_dict is None:
+        input_dict = {}
+
+    return vol.Schema(
+        {
+            vol.Required(
+                CONF_NAME, default=input_dict.get(CONF_NAME, DEFAULT_NAME)
+            ): str,
+            vol.Required(CONF_API_KEY, default=input_dict.get(CONF_API_KEY)): str,
+            vol.Inclusive(
+                CONF_LATITUDE,
+                "location",
+                default=input_dict.get(CONF_LATITUDE, hass.config.latitude),
+            ): cv.latitude,
+            vol.Inclusive(
+                CONF_LONGITUDE,
+                "location",
+                default=input_dict.get(CONF_LONGITUDE, hass.config.longitude),
+            ): cv.longitude,
+        },
+        extra=vol.REMOVE_EXTRA,
+    )
+
+
+def _get_unique_id(hass: HomeAssistantType, input_dict: Dict[str, Any]):
+    """Return unique ID from config data."""
+    return (
+        f"{input_dict[CONF_API_KEY]}"
+        f"_{input_dict.get(CONF_LATITUDE, hass.config.latitude)}"
+        f"_{input_dict.get(CONF_LONGITUDE, hass.config.longitude)}"
+    )
+
+
+class ClimaCellOptionsConfigFlow(config_entries.OptionsFlow):
+    """Handle ClimaCell options."""
+
+    def __init__(self, config_entry: config_entries.ConfigEntry) -> None:
+        """Initialize ClimaCell options flow."""
+        self._config_entry = config_entry
+
+    async def async_step_init(
+        self, user_input: Dict[str, Any] = None
+    ) -> Dict[str, Any]:
+        """Manage the ClimaCell options."""
+        if user_input is not None:
+            return self.async_create_entry(title="", data=user_input)
+
+        options_schema = {
+            vol.Optional(
+                CONF_FORECAST_TYPES,
```

Comment:
I think we can remove the forecast type option and just let the user enable/disable the entities in the entity registry. Enable the default forecast type by default and disable the other ones.

Sorry for not mentioning this earlier. I was reminded recently about this by Matthew, code owner for NWS.

Reply:
no worries, I must have missed this when I switched to disabling entities by default. Just trying to get this guy closed out 😄